### PR TITLE
Round Robin Assignment

### PR DESCRIPTION
**Pull request by LX5321**
Added code to find assignee usernames in the commit body or assign a single assignee if no assignee is mentioned.

### CODE REVIEW

**Comment by LX5321** [comment]
This PR resolves #10

File: src/logic.py [hunk 1]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
             True,
             f'Author {self.author} successfully assigned to PR, "{self.config.awaiting_update_label}" label added',
         )
+    
+    def parse_magic_string(self) -> str:
+        """
+            Parses the commit body to find magic string
```
Comment (samuelcolvin) [inline src/logic.py:183]:
```suggestion
            Parses the commit body to find magic string
```
Please fix the indent, we should be looking for the magic string in the PR body, not the commit message.

File: src/logic.py [hunk 2]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
...
+        """
+
+        _username = None
```
Comment (samuelcolvin) [inline src/logic.py:186]: you don't need this.

File: src/logic.py [hunk 3]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
...
+        search_username = re.findall(r'primary-reviewer:\s@[a-zA-Z][a-zA-Z0-9-_\.]{1,20}$', self.comment.body)
```
Comment (samuelcolvin) [inline src/logic.py:187]: use `re.search`, not `re.findall`.

File: src/logic.py [hunk 4]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
...
+
+
+        if(len(search_username)):
```
Comment (samuelcolvin) [inline src/logic.py:190]: you don't need the outer brackets, you don't need `len(...)` for boolean checks on lists, if you use `re.search` you'll get back an optional match.

File: src/logic.py [hunk 5]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
...
+            # found magic comment
+            _username = search_username[0].split('@')[1]
+        else:
+            _username = random.choice(self.reviewers)
+            self.comment.body += f"\n\n{_username}"
```
Comment (samuelcolvin) [inline src/logic.py:195]: this doesn't update anything.

Comment (samuelcolvin) [reply src/logic.py:195]: also the appended string needs to match the magic format.

File: src/logic.py [hunk 6]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
...
+
+        _username = None
+        search_username = re.findall(r'primary-reviewer:\s@[a-zA-Z][a-zA-Z0-9-_\.]{1,20}$', self.comment.body)
+
+
+        if(len(search_username)):
+            # found magic comment
+            _username = search_username[0].split('@')[1]
+        else:
+            _username = random.choice(self.reviewers)
```
Comment (samuelcolvin) [inline src/logic.py:194]: best if we can use round robin by storing the last used reviewer in redis.

File: src/logic.py [hunk 7]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
...
+            Parses the commit body to find magic string
+        """
+
+        _username = None
+        search_username = re.findall(r'primary-reviewer:\s@[a-zA-Z][a-zA-Z0-9-_\.]{1,20}$', self.comment.body)
+
+
+        if(len(search_username)):
+            # found magic comment
+            _username = search_username[0].split('@')[1]
```
Comment (samuelcolvin) [inline src/logic.py:192]: no need for an underscore in the var name.

File: src/logic.py [hunk 8]
```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
...
+        else:
+            _username = random.choice(self.reviewers)
+            self.comment.body += f"\n\n{_username}"
+        
+        return _username
```
Comment (samuelcolvin) [inline src/logic.py:197]: you need to check the reviewer is in `self.reviewers` and return an error if not.

**Comment by LX5321** [comment]
Hey @samuelcolvin thanks for the review. Will fix and get back ASAP.

**Comment by LX5321** [comment]
Will work on refactoring the test cases.

**Comment by LX5321** [comment]
Hey @samuelcolvin requesting you to please check the test cases that I modified in this PR.
I'm not quite comfortable on the unit test cases part.

Do let me know the steps that I can take if any.

Thanks!

**Comment by samuelcolvin** [comment]
@LX5321 thanks so much for your work on this, I've made some changes to get the PR ready to be merged.

**Comment by LX5321** [comment]
Greetings @samuelcolvin was just going through the changes that you made. I look forward to being able to be write code like this in the future.

Thanks for the suggestions and help on this PR!

**Pull request merged**